controller/lib/hal/hal.h

```diff
@@ -28,12 +28,34 @@
 // when compiling for test, but not when compiling for production.
 
 #ifdef TEST_MODE
+
+#ifdef AVR
+#error "TEST_MODE intended to be run only on native, but AVR is defined"
+#endif
+
 #include "gmock/gmock.h"
 #define HAL_MOCK_METHOD(returntype, name, args)                                \
   MOCK_METHOD(returntype, name, args)
-#else
-#define HAL_MOCK_METHOD(returntype, name, args) returntype name args
+#define HAL_CONSTANT(name) HAL_##name
+
+#else // !TEST_MODE
+
+#ifndef AVR
+#error "When running without TEST_MODE, expecting AVR to be defined"
 #endif
+#include <Arduino.h>
+
+#define HAL_MOCK_METHOD(returntype, name, args) returntype name args
+#define HAL_CONSTANT(name) HAL_##name = name
```

Comment:
Since it's an enum *class*, I wonder if `name = name` or maybe `name = ::##name` would DTRT.  Might make it slightly nicer to use this interface.

Reply:
I had to do it this way because Arduino's HIGH, LOW etc are macros, so "HIGH = HIGH" would e.g. expand to "1 = 1".

Reply:
smh